chapter/evaluation.md

```diff
@@ -0,0 +1,158 @@
+[Section @sec:results] summarises the results of these methods.
+
+## Evaluation Considerations
+
+Different methods to evaluate the abovementioned metrics were considered.
+While quantifying user experience yields statistically sound insights about the studied subject, it fails to point out specific user needs.
+Therefore, this work employs a more subjective evaluation based on a standardized experience report focusing on individual features.
+Contrasting the expectations highlights well executed, immature or missing features.
+This allows more actionable planning of the future development to meet user expectations.
+
+On the other hand it is more approachable to track runtime performance objectively through time measurements.
+In fact, runtime behavior was a central assumption underlying the server architecture. 
+As discussed in [@sec:considerations] NLS follows an eager, non-incremental processing model.
+While incremental implementations are more efficient, as they do not require entire files to be updated, they require explicit language support, i.e., an incremental parser and analysis.
+Implementing these functions exceeds the scope of this work.
+Choosing a non-incremental model on the other hand allowed to reuse entire modules of the Nickel language.
+The analysis itself can be implemented both in a lazy or eager fashion.
+Lazy analysis implies that the majority of information is resolved only upon request instead of ahead of time.
+That is, an LSP request is delayed by the analysis before a response is made.
+Some lazy models also support memoizing requests, avoiding to recompute previously requested values.
+However, eager approaches preprocess the file ahead of time and store the analysis results such that requests can be handled mostly through value lookups.
+To fit Nickels' type-checking model and considering that in a typical Nickel workflow, the analysis should still be reasonably efficient, the eager processing model was chosen over a lazy one.
+
+## Methods
+
+### Objectives
+
+The qualitative evaluation was conducted with a strong focus on the first metric in [sec:metrics].
+Usability proves hard to quantify, as it is tightly connected to subjective perception, expectations and tolerances.
+The structure of the survey is guided by two additional objectives, endorsing the separation of individual features.
+On one hand, the survey should inform the future development of NLS; which feature has to be improved, which bugs exist, what do users expect.
+This data is important as for NLS both as an LSP implementation for Nickel (affecting the perceived maturity of Nickel) as well as a generic basis for other projects. 
+On the other hand, all features are implemented on top of the same base (cf. [@sec:implementation]).
```

Comment:
```suggestion
On the other hand, since all features are essentially queries to the common linearization data structure (cf. [@sec:implementation), the implementation of this central structure is an essential consideration.```